tests/net/lib/dns_sd/src/main.c

```diff
@@ -683,6 +681,69 @@ static void test_setup_dst_addr(void)
 		      setup_dst_addr(ctx_xx, pkt_xx, &dst, &dst_len), "");
 }
 
+/** test for @ref dns_sd_is_service_type_enumeration */
+static void test_is_service_type_enumeration(void)
+{
+	static const struct dns_sd_rec filter_ok = {
+		.instance = "_services",
+		.service = "_dns-sd",
+		.proto = "_udp",
+		/* TODO: support additional service domains */
+		.domain = "local",
+		.text = dns_sd_empty_txt,
+		.text_size = sizeof(dns_sd_empty_txt),
+		.port = &dns_sd_port_zero,
+	};
+
+	zassert_true(dns_sd_is_service_type_enumeration(&filter_ok), "");
+
+	static const struct dns_sd_rec filter_nok = {
+		/* not a service_type_enumeration */
+		.instance = "_serv1c3s",   .service = "_dns-sd",
+		.proto = "_udp",	   .domain = "local",
+		.text = dns_sd_empty_txt,  .text_size = sizeof(dns_sd_empty_txt),
+		.port = &dns_sd_port_zero,
+	};
+
+	zassert_false(dns_sd_is_service_type_enumeration(&filter_nok), "");
+}
+
+static void test_extract_service_type_enumeration(void)
+{
+	static const uint8_t query[] = {
+		0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0x18, 0x00, 0x00, 0x00, 0x00, 0x09, 0x5f,
```

Comment:
These numbers come from the query for `_services._dns-sd._udp.local`
<img width="1552" alt="Screen Shot 2021-09-23 at 7 15 02 AM" src="https://user-images.githubusercontent.com/983494/134788034-e23cd149-e0c3-4197-8d0a-badc31d24bef.png">